packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -38,28 +40,59 @@ class ChartConnection extends DisposableController
   Timer? _pollingTimer;
   bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If true, the connection to application was lost.
+  bool _disconnected = false;
+
+  late final isDeviceAndroid = _isDevToolsCurrentlyConnected()
```

Comment:
this can be 
`bool get isDeviceAndroid => serviceConnection.serviceManager.vm?.operatingSystem == 'android';`

This will return false if vm is null, so no need to manually check

Reply:
1. If globals[ServiceConnectionManager] is null this will fail.
2. This value does not change during lifetime of the instance, so there is no need to detect it every time, with risk of failure at disconnect

Reply:
Okay, then we can still make this more verbose but leave as late final:

`late final isDeviceAndroid = serviceConnection.serviceManager.vm?.operatingSystem == 'android';`